Sources/MuxUploadSDK/InternalUtilities/ChunkedFile.swift

```diff
@@ -39,11 +39,21 @@ class ChunkedFile {
             return Result.failure(ChunkedFileError.fileHandle(error))
         }
     }
+
+    func fetchFileSize(
+        fileURL: URL
+    ) throws -> UInt64 {
+        guard let fileSize = try FileManager.default.attributesOfItem(atPath: fileURL.path)[FileAttributeKey.size] as? UInt64 else {
```

Comment:
thought(non-blocking): given that there are multiple places where we appear to need the `fileSize`, it may make sense to "memoize" the value with `_fileSize` if `ChunkedFile` is intended to have a 1:1 correspondence with a file. Currently, part of the implementation is more like a "bundle of (pure) utility functions" and part is "an instance modeling state related to a particular file". Might be good to pin down which we're aiming for here.

Reply:
I think reading the file size from the relevant source of truth (the iOS file manager) makes more sense here. Caching it seems like premature optimization to me.

Reply:
Yup definitely non-blocking. That said, if under any condition the value of the file size reported from the file manager changed under foot mid-upload (say, some other process modified the file), wouldn't that cause issues here? Relates to @daytime-em's comment elsewhere.

Reply:
Yes if the file itself was changed mid-upload, the upload would experience issues as @daytime-em mentioned above. That however is affects the upload irrespective of how the file size value is stored or accessed, seems like an unrelated issue.

Reply:
We're planting the seeds of a possible bug here, unless we also back the property `ChunkedFile.fileSize` by a return value from this new `fetchFileSize()`, which I *thiiink* is also Christian's point?

Reply:
My thinking is that `ChunkedFile.fileSize` should be removed entirely, anything accessing that property should just use `FileManager` (the iOS Foundation API for accessing file metadata.) I believe this approach prevents potential several potential bugs.

Reply:
I don't see any benefit in wrapping calls to the Foundation `FileManager` in a static function or a general utility class.

Reply:
> I believe this approach prevents several potential bugs.

I think that's very possible, but in order to obtain the most benefit from this approach I think you need to carry it all the way through. Luckily this is easy because `fileSize` is a computed property. Changing it to do what you described (fetching the size of the file `ChunkedFile` is working with) is [pretty simple](https://github.com/muxinc/swift-upload-sdk/commit/7f3af2a161bb47b147cc07c223df2d71976cf149)

Reply:
.. And I still request that you make `fetchFileSize` static/a class method. The instance methods all operate on the file the instance is managing. 

And I think I understand why you consider `openFile` to be a side effect now. But `ChunkedFile` isn't a collection of unrelated utility methods, it's a single object that exposes operations. The internal bookkeeping that `openFile` does is part of the monad/context/object state.

If we wish to refactor `ChunkedFile` to be utility methods with no internal state, we should do that separately or else we wind up with a class that takes both approaches badly.

Reply:
Yup I have all references `ChunkedFile.fileSize` removed on my local branch, testing now to make sure uploads behave as expected. Will update this PR once I'm done testing.

Am resolving this thread but feel free to reopen or add new comments relating to the upcoming set of changes once they are up.

Reply:
> .. And I still request that you make `fetchFileSize` static/a class method. The instance methods all operate on the file the instance is managing.
> 
> And I think I understand why you consider `openFile` to be a side effect now. But `ChunkedFile` isn't a collection of unrelated utility methods, it's a single object that exposes operations on the state of the object. The internal bookkeeping that `openFile` does is part of the monad/context/object state.
> 
> If we wish to refactor `ChunkedFile` to be utility methods with no internal state, we should do that separately or else we wind up with a class that takes both approaches badly.

Happy to discuss this once my next set of changes is up, but currently `fetchFileSize` isn't there any longer.